ndcube/mixins/plotting.py

```diff
@@ -138,11 +138,22 @@ def _plot_1D_cube(self, wcs, axes=None, axes_coordinates=None, axes_units=None,
             if yerror is not None:
                 yerror = np.ma.masked_array(yerror, self.mask)
 
+        # The centers of the bins are integer values, so the edges are at 0.5
+        x = np.arange(ydata.size + 1) - 0.5
+        # Need to append the final value to get `axes.step` to draw the final
+        # bin. When NDCube depends on matplotlib>=3.4, `axes.stairs` can be
+        # used instead to avoid this extra step.
+        ydata = np.append(ydata, ydata[-1])
+
+        lines = axes.step(x, ydata, where='post', **kwargs)
         if yerror is not None:
-            # We plot against pixel coordinates
-            axes.errorbar(np.arange(len(ydata)), ydata, yerr=yerror, **kwargs)
```

Comment:
If we implement this, I think this still the right line of code for the step case, except that your should add `fmt='none'` so that only the errorbars are plotted.